### PR TITLE
Species job blacklist

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Adds the ability to restrict species from holding certain roles. Diona who can become security is stupid, no?

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image

**Changelog**

### CODE REVIEW

File: Content.Client/LateJoin/LateJoinGui.cs [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 using Content.Client.GameTicking.Managers;
+using Content.Client.Humanoid;
 using Content.Client.UserInterface.Controls;
 using Content.Client.Players.PlayTimeTracking;
+using Content.Client.Preferences;
 using Content.Shared.CCVar;
+using Content.Shared.Humanoid.Prototypes;
+using Content.Shared.Preferences;
 using Content.Shared.Roles;
+using NFluidsynth;
```
Comment (metalgearsloth) [inline Content.Client/LateJoin/LateJoinGui.cs:12]: Wrong logs.

**Comment by Aisu9** [comment]
i mean it does make sense... but what about speciesism being illegal(and banable) in this version of Nanotrasen/in SS14

**Comment by LankLTE** [comment]
> i mean it does make sense... but what about speciesism being illegal/outlawed(and banable) in this version of Nanotrasen/in SS14

This is for downstream (and disabled by default I believe). Won't be used in WizDen

**Comment by VigersRay** [comment]
> > i mean it does make sense... but what about speciesism being illegal/outlawed(and banable) in this version of Nanotrasen/in SS14
> 
> This is for downstream (and disabled by default I believe). Won't be used in WizDen

quite right

**Comment by KasianenkoViacheslavMiki** [comment]
Sorry, what I write in this conversation and so not a contributor. But how I think and wrote EmoGarbage404 is better in a Job's prototype. But not as Job Requirement, but as a different component. For example how I created a similar feature.
https://github.com/KasianenkoViacheslavMiki/space-station-14/commit/21977197fc71702f444c905095d15727e63b3457 
So you can simply configure in Job's prototype which species must be banned for this job, which is more logical.

**Comment by VigersRay** [comment]
> Sorry, what I write in this conversation and so not a contributor. But how I think and wrote EmoGarbage404 is better in a Job's prototype. But not as Job Requirement, but as a different component. For example how I created a similar feature.
> https://github.com/KasianenkoViacheslavMiki/space-station-14/commit/21977197fc71702f444c905095d15727e63b3457 
> So you can simply configure in Job's prototype which species must be banned for this job, which is more logical. 

Well, yes, this is the best option, I was just too lazy to specify a ban for the species for 20 roles in my build, so I decided that it would be easier to specify them in the species component

**Comment by Emisse** [comment]
i dont think putting this on the species is a good option, having it be a blacklist on the job prototype would be a lot better and save you headache in the future

**Comment by VigersRay** [comment]
> i dont think putting this on the species is a good option, having it be a blacklist on the job prototype would be a lot better and save you headache in the future

Yeah, it's probably better that way, I'll move it there when I have free time.

**Comment by VigersRay** [comment]
> i dont think putting this on the species is a good option, having it be a blacklist on the job prototype would be a lot better and save you headache in the future

Yeah, it's probably better that way, I'll move it there when I have free time.

**Comment by VigersRay** [comment]
> > Diona who can become security is stupid, no?
> 
> On upstream diona can harm, so it's not stupid.

This was just an example, in the code I did not apply restrictions for any of the races, each individual build will decide for itself which restrictions need to be applied.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
As emisse said this should be a jobrequirement.

**Comment by metalgearsloth** [comment]
@VigersRay are you coming back to this

**Pull request closed without merging**